### DISCUSSION TITLE
Filter Webpart showing No Data Error after mapping SiteName to RefinableString

### DISCUSSION THREAD

**Discussion by maj-jacob** [microsoft-search/pnp-modern-search#3253] [Q&A] [answered]
**Version used**
4.9

**Describe the bug**
After mapping the crawled property _ows_SiteName_ to a managed property _RefinableString00_ to use it in the filter webpart, it is not possible to configure a new filter webpart. It shows the error "no data in your collection". 
When there was a filter webpart configured before the mapping it is still working, but it is not possible to add new Filter fields to this webpart.
After unmapping _ows_SiteName_ from _RefinableString00_ the behaviour stays the same.

**To Reproduce**
Detailed steps to reproduce the behavior:
**I recommend doing this on a non productive tenant**
1. map _ows_SiteName_ to a managed property e.g. _RefinableString00_
2. trigger mapping process by rename a site
4. add PnP Search Result and PnP Filter Webpart on one Page an connect them
5. select the managed property in the search results configuration
6. edit Filters Webpart
7. Filter Webpart should show "no data in your collection" (see screenshot) and no filtering should not be possible

**Expected behavior**
The filter webpart should be working.

**Screenshots**
![grafik](https://github.com/microsoft-search/pnp-modern-search/assets/139750982/5d004d2c-5ace-45d1-bc6d-9f340fc06ca6)

**Desktop (please complete the following information):**
 all browsers

**Additional context**
there are not issues with the search results themselfs, I reproduced the behaviour on two tenants

**Comment by kasperbolarsen** [comment]
Could this be similar to https://github.com/microsoft-search/pnp-modern-search/issues/3237 ? That the query has not been set yet?

**Comment by maj-jacob** [comment]
> Could this be similar to #3237 ? That the query has not been set yet?

thank you for your suggestion. The step is mentioned in step 4 in the repoduction section. also in general not connecting the query is not causing this error in the filter webpart. the error "no data in your collection" in this issue appears in any case. even if you do not want to use the managed property in the filter webpart. any new filter webpart shows the error and any existing one cannot be extended with any new filter fields.

**Comment by wobba** [comment]
I believe editing a site title might not immediately re-index all items and have it populated. I would hit "re-index this site" and wait. Or create a new item to test. I happened to have the same mapping myself tenant wide and using this brings site names back just fine.
![image](https://github.com/microsoft-search/pnp-modern-search/assets/51104/08a7c6bd-b419-4fba-bf71-9c9fb6e0d77b)

![image](https://github.com/microsoft-search/pnp-modern-search/assets/51104/bc829dbe-165c-40d3-9ad0-1516b92751cd)

**Reply by maj-jacob** [reply]
The mapping itself comes back fine after a  re-crawl - also in the Result Webpart. it's the Filter Webpart that does not work anymore.

![grafik](https://github.com/microsoft-search/pnp-modern-search/assets/139750982/9a53b470-27b7-42ec-9179-75b0b3363451)

**Reply by maj-jacob** [reply]
@wobba I re-indexed one site just to try everything but as expected it did nothing to the broken filter webpart. Are you able to use the filter webpart after this specific mapping? ows_sitename <> refinablestringXX  ? I broke the PnP Modern Search Webparts on two tenants with this which worked perfectly before that.

**Reply by wobba** [reply]
The filter web part works just fine for me. If you have connected the filter web part you should also see in the request that it asks file RefinableString 01 as a filter, and you should see it come back in the response, also in the debug template.

**Comment by maj-jacob** [comment, marked as answer]
It turned out that it was a general issue with the MS Search. It involved also the vertials in the MS Modern Search. They did something - wouldn't tell me what exactly - that fixed both. Thank y'all for your replies!
